packages/uilib/src/lib/plugins/communication-explorer/sidebar/sidebar.svelte

```diff
@@ -57,13 +58,30 @@
     	const target = e.target as HTMLInputElement
     	setNameFilter(target.value)
     }
+
+    let searchQuery = "";
+    $: filteredNodes = rootNode.children.filter((node) =>
+        node.label.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
+
+    function handleIEDSearch(e: Event) {
+        IEDSelectionIDs = [""]; 
+        const target = e.target as HTMLInputElement;
+        searchQuery = target.value;
+    }
+
+    function clearAll() {
+        searchQuery = "";
+        clearSelection();
+    }
+
 </script>
 
 <div class="sidebar sidebar-right">
     <div class="sidebar-content">
         <!-- svelte-ignore a11y-missing-attribute -->
         <div class="actions">
-            <a class="clear-all" on:keypress on:click={clearSelection}>
+            <a class="clear-all" on:keypress on:click={clearAll}>
```

Comment:
an `a` attribute for a button is not advised for accessibility reasons. I think it could be replaced by a real button element, also through a svelte MUI component for design consistency.

Reply:
a:
![image](https://github.com/user-attachments/assets/47a35062-a584-4740-b49a-0cd426695681)

button:
![Bildschirmfoto 2024-10-09 um 10 54 57](https://github.com/user-attachments/assets/f5eabcaf-4084-4767-b08e-29448bf830ca)

Yes, but should be adapted again by UX. In my view, this is more the task of a major UX overhaul of the sidebar.